test/powershell/Modules/Microsoft.PowerShell.Management/Set-Service.Tests.ps1

```diff
@@ -235,6 +235,85 @@ Describe "Set/New/Remove-Service cmdlet tests" -Tags "Feature", "RequireAdminOnW
         { Remove-Service -Name "testremoveservice" -ErrorAction 'Stop' } | ShouldBeErrorId "InvalidOperationException,Microsoft.PowerShell.Commands.RemoveServiceCommand"
     }
 
+    It "Get-Service can get the username of a service" {
+        try {
+            $userName = "user1"
```

Comment:
Because there's very similar code for these test cases, you could turn them into `-TestCases` like:

```powershell

# move user creation to BeforeAll{} and cleanup in AfterAll{}

It "Get-Service can get the <property> of a service" -TestCases @(
  @{property="username"; value="user1"; parameters = @{ Credential = $creds },
  @{property="binpath";value="$PSHOME\powershell.exe"},
 ...
){
 param ($property, $value, $parameters)

  if ($parameters -eq $null) {
    $parameters = @{}
  }
  $parameters += @{Name=$servicename;binarypathname="...}
  $service = New-Service @parameters
  ...
  $service.$property | Should BeExactly $value
}
```